### PR TITLE
ci: scope each tests/examples invocation to a specific crate

### PR DESCRIPTION
**Pull request by ipetkov**
## Motivation
For each cross product of a crate and it's features, the CI is running all tests for all crates.

## Solution
Scope each test invocation to a specific crate's directory, so only its tests run.

### CODE REVIEW

**Comment by carllerche** [comment]
LGTM to me thanks. I can't +1 yet though :)

**Comment by ipetkov** [comment]
Ugh I hate yaml, I think something is going wrong with the nested parameters and the value isn't being picked up by azure pipelines

**Comment by ipetkov** [comment]
Whew finally got it all working together. Basically the main constraints are:

- cargo doesn't support specifying a "default" feature unless its explicitly defined in the `Cargo.toml`
- cargo doesn't support specifying `--features` with an empty parameter (an empty string should work but then we have to wrangle quoting issues across yaml and the CLI)
- yaml needs us to specify an actual empty array as the value for the crate name (otherwise it uses null which fails in the for-each loop)
  - but we also can't simply put an empty `-` line after the crate name because that resolves to an array with a null element which also causes problems

I hate writing ugly yaml as much as the next person, so totally open to devising a different scheme for specifying our tests (at least now we have a working solution to work from!)

**Comment by ipetkov** [comment]
Thanks for the review @LucioFranco, I'm going to merge this in, we can iterate more on it later :)

File: azure-pipelines.yml [hunk 1]
```diff
@@ -49,24 +47,17 @@ jobs:
...
       # - tokio-buf
-      tokio-codec:
-        - default
-      tokio-current-thread:
-        - default
-      tokio-executor:
-        - default
-      tokio-io:
-        - default
+      tokio-codec: []
```
Comment (carllerche) [inline azure-pipelines.yml]: Ok, I didn't know about `foo: []` syntax... much better, thanks :+1:

**Pull request merged**